### ISSUE TITLE
Consider using browserify or webpack

### ISSUE THREAD

**Issue by ChALkeR** [qmlweb/qmlweb#45]
Atm, we bundle `uglify-js` is `src` dir and have our own encapsulate wrappers. That is affecting several things and has to be fixed in some way, and using browserify or webpack to make sure that we don't have to deal manually with various module systems would be nice.

**Comment by ChALkeR** [comment]
This partially blocks #45 (I could use a work-around, but that would be ugly).

**Comment by ChALkeR** [comment]
This also should be the way to fix #41.

**Comment by henrikrudstrom** [comment]
my vote goes for Browserify, i understand Webpack doesnt play well with npm

**Comment by Plaristote** [comment]
I've looked it up as well, and I lean towards Browserify as well.

**Comment by aspidites** [comment]
We use web pack at work so I have a bit of experience with it. I'm not sure if browserify has equivalents, but I really like the following for webpack:
 - hot module reloading
 - Automatic bundling of common functionality between modules
- loaders for various precompilers
- dev server

**Comment by ChALkeR** [comment]
I don't think that we need to over-complicate things, after all.
It should be good as it is now.

We should support those, though, but that's a separate matter.

**Comment by ChALkeR** [comment]
I am going to close this.

I proposed this mainly for these reasons:
 * We were using uglify-js and needed a better way to load it.
 * I thought that it would help us with coverage analysis, by moving the build stage (#40, #66).
 * I thought that this would help us to split the parser (#41) and move it to a separate repo (#173).
 * I thought that this would help us to solve cross-element dependencies in an automatic way.

All of those are fixed now with simplier methods.

Feel free to reopen if you want to continue the discussion, though.

**Comment by henrikrudstrom** [comment]
I would like to open this again. one thing that isnt addressed by the recent changes is using the qml engine in node.js environment, might be a bit of a niche application, but incidentally thats something that i need right now, so i'd be happy to do the work!

**Comment by ChALkeR** [comment]
@henrikrudstrom Ah, fine, thanks for reopening!

Could you provide more details on the usecase? I was under an impression that only `parser` is needed in a non-browser enviroment now.

**Comment by henrikrudstrom** [comment]
Im going to use the declarative engine to define solid 3d models, using a c++ library, so I need to be able to run it server side or in a nwjs application.

**Comment by Plaristote** [comment]
True, we need to properly export our stuff so that it can be required from node.js.
We haven't really done that, have we ?

**Comment by henrikrudstrom** [comment]
@Plaristote nope, your browserify branch does that, at least some of functions. Im thinking to modify that branch: add index.js and require all the files from there (instead of concatenating them to qt.js)

**Comment by ChALkeR** [comment]
Not very active, let's postpone this. I'm not closing, though.
